### PR TITLE
Bump to richie 2.2.0 and prepare releases

### PR DESCRIPTION
**Pull request by jbpenrath**
### Added

- Use custom views to handle errors (400, 403, 404, 500)
- Enable LTIConsumerPlugin on `course_teaser` placeholder

### Added

- Use custom views to handle errors (400, 403, 404, 500)
- Enable LTIConsumerPlugin on `course_teaser` placeholder

### Added

- Add "Types" category as search filter
- Use custom views to handle errors (400, 403, 404, 500)
- Enable LTIConsumerPlugin on `course_teaser` placeholder

### Changed

- Prevent course run languages update when synchronization hook is triggered
- Set english language as public

### Added

- Use custom views to handle errors (400, 403, 404, 500)
- Enable LTIConsumerPlugin on `course_teaser` placeholder

### Added

- Use custom views to handle errors (400, 403, 404, 500)
- Enable LTIConsumerPlugin on `course_teaser` placeholder

### CODE REVIEW

**Review by kernicPanel — APPROVED** [review]
OK

**Comment by sampaccoud** [comment]
@jbpenrath we must also reactive the english language on fun-mooc. They need to see the public version to ensure how it works and will finalize it.

File: sites/cnfpt/src/backend/locale/fr/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -21,32 +21,52 @@ msgstr ""
 msgid "Unsorted uploads are not allowed."
 msgstr "Les téléchargements non classés sont interdits"
 
-#: cnfpt/settings.py:210
+#: cnfpt/settings.py:213
+msgid "Dashboard"
+msgstr ""
```
Comment (sampaccoud) [inline sites/cnfpt/src/backend/locale/fr/LC_MESSAGES/django.po:26]: translate to "Tableau de bord"?

By the way I'm not sure why these changes pop-up and only in fr language?

File: sites/cnfpt/src/backend/locale/fr/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -21,32 +21,52 @@ msgstr ""
...
+
+#: cnfpt/settings.py:214
+msgid "{base_url:s}/dashboard"
```
Comment (sampaccoud) [inline sites/cnfpt/src/backend/locale/fr/LC_MESSAGES/django.po:29]: ditto

File: sites/funcorporate/src/backend/funcorporate/settings.py [hunk 3]
```diff
@@ -437,6 +437,7 @@ class Base(StyleguideMixin, DRFMixin, RichieCoursesConfigurationMixin, Configura
     RICHIE_ES_INDICES_PREFIX = values.Value(
         default="richie", environ_name="RICHIE_ES_INDICES_PREFIX", environ_prefix=None
     )
+    RICHIE_ES_STATE_WEIGHTS = values.ListValue(None)
```
Comment (sampaccoud) [inline sites/funcorporate/src/backend/funcorporate/settings.py:441]: are you missing teaser placeholder changes?

File: sites/funmooc/src/backend/locale/fr/LC_MESSAGES/django.po [hunk 4]
```diff
@@ -21,51 +21,71 @@ msgstr ""
 msgid "Unsorted uploads are not allowed."
 msgstr "Les téléchargements non classés sont interdits"
 
-#: funmooc/settings.py:219
+#: funmooc/settings.py:222
+msgid "Dashboard"
+msgstr ""
```
Comment (sampaccoud) [inline sites/funmooc/src/backend/locale/fr/LC_MESSAGES/django.po:26]: translate strings in this file?

File: sites/funmooc/src/backend/locale/fr/LC_MESSAGES/django.po [hunk 5]
```diff
@@ -21,51 +21,71 @@ msgstr ""
...
+msgid "{base_url:s}/dashboard"
+msgstr ""
+
+#: funmooc/settings.py:226
 msgid "Profile"
 msgstr "Profil"
 
-#: funmooc/settings.py:220
+#: funmooc/settings.py:227
+msgid "{base_url:s}/u/(username)"
```
Comment (sampaccoud) [inline sites/funmooc/src/backend/locale/fr/LC_MESSAGES/django.po:37]: this pattern looks weird (username)?

File: sites/cnfpt/CHANGELOG.md [hunk 6]
```diff
@@ -8,8 +8,12 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
 
+## [1.1.0] - 2021-03-05
+
 ### Added
+
 - Use custom views to handle errors (400, 403, 404, 500)
+- Enable LTIConsumerPlugin on `course_teaser` placeholder
```
Comment (sampaccoud) [inline sites/cnfpt/CHANGELOG.md:16]: You missed this settings: https://github.com/openfun/richie/blob/master/sandbox/settings.py#L319

I propose configuring the LTI consumers directly in the code here and leave on the secrets to matsuo... (for the moment we have only one LTI service to configure: marsha).

File: sites/cnfpt/src/backend/locale/fr/LC_MESSAGES/django.po (hunk 1 above)
Comment (jbpenrath) [reply sites/cnfpt/src/backend/locale/fr/LC_MESSAGES/django.po:26]: Because for CNFPT this is the only one language and there is just an offset of this translation.

File: sites/funmooc/src/backend/locale/fr/LC_MESSAGES/django.po (hunk 5 above)
Comment (jbpenrath) [reply sites/funmooc/src/backend/locale/fr/LC_MESSAGES/django.po:37]: This is something use by the frontend to be able to dynamically replace `(username)` string by the authenticated username.

File: sites/funcorporate/src/backend/funcorporate/settings.py (hunk 3 above)
Comment (jbpenrath) [reply sites/funcorporate/src/backend/funcorporate/settings.py:441]: Whoops!

File: sites/cnfpt/src/backend/locale/fr/LC_MESSAGES/django.po (hunk 2 above)
Comment (jbpenrath) [reply sites/cnfpt/src/backend/locale/fr/LC_MESSAGES/django.po:29]: On OpenEdX this url does not change according to the language.

File: sites/cnfpt/src/backend/cnfpt/settings.py [hunk 7]
```diff
@@ -464,6 +471,32 @@ class Base(StyleguideMixin, DRFMixin, RichieCoursesConfigurationMixin, Configura
...
+                "InsecureSharedSecret",
+                environ_name="LTI_TEST_SHARED_SECRET",
+                environ_prefix=None,
+            ),
+            "base_url": values.Value(
+                "https://lti.tools/saltire/tp",
+                environ_name="LTI_TEST_BASE_URL",
+                environ_prefix=None,
+            ),
+            "display_name": "LTI Provider Marsha Video",
```
Comment (sampaccoud) [inline sites/cnfpt/src/backend/cnfpt/settings.py:494]:
```suggestion
            "display_name": "Marsha Video",
```

File: sites/cnfpt/src/backend/cnfpt/settings.py [hunk 8]
```diff
@@ -464,6 +471,32 @@ class Base(StyleguideMixin, DRFMixin, RichieCoursesConfigurationMixin, Configura
...
+                environ_name="LTI_TEST_OAUTH_CONSUMER_KEY",
+                environ_prefix=None,
+            ),
+            "shared_secret": values.Value(
+                "InsecureSharedSecret",
+                environ_name="LTI_TEST_SHARED_SECRET",
+                environ_prefix=None,
+            ),
+            "base_url": values.Value(
+                "https://lti.tools/saltire/tp",
```
Comment (sampaccoud) [inline sites/cnfpt/src/backend/cnfpt/settings.py:490]: since you put the marsha name, put the marsha url pattern:
`https://marsha\.education/lti/videos/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}`

File: sites/funcorporate/src/backend/funcorporate/settings.py [hunk 9]
```diff
@@ -338,9 +338,14 @@ class Base(StyleguideMixin, DRFMixin, RichieCoursesConfigurationMixin, Configura
 
     # Placeholders
     CMS_PLACEHOLDER_CONF_OVERRIDES = {
+        "courses/cms/course_detail.html course_teaser": {
+            "name": _("Teaser"),
+            "plugins": ["VideoPlayerPlugin", "LTIConsumerPlugin"],
+            "limits": {"VideoPlayerPlugin": 1, "LTIConsumerPlugin": 1},
+        },
```
Comment (sampaccoud) [inline sites/funcorporate/src/backend/funcorporate/settings.py:345]: You forgot the merge dict part

Comment (jbpenrath) [reply sites/funcorporate/src/backend/funcorporate/settings.py:345]: 😬

**Review by sampaccoud — APPROVED** [review]
